include/test/mir/test/doubles/file_logger.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2022 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_TEST_DOUBLES_FILE_LOGGER_H_
+#define MIR_TEST_DOUBLES_FILE_LOGGER_H_
+
+#include "mir/logging/logger.h"
+
+#include <fstream>
+#include <string>
+
+namespace mir
+{
+namespace test
+{
+namespace doubles
+{
+class FileLogger : public mir::logging::Logger
+{
+public:
+    FileLogger(const std::string& filename, bool std_too);
```

Comment:
This signature seems somewhat odd and limiting.

1. Should this constructor be responsible for opening the file? Or should an `ofstream` be moved in? (That allows the output stream to be constructed in other ways)
2. The `std_too` flag mostly replicates `DumbConsoleLogger` - that duplication is unnecessary. (And what about `glog_too`?)
3. Shouldn't it be possible to log different severities to different files?

Re 2 I think _Composite_ pattern would be a better way of combining multiple logging sinks.

Reply:
2. Resolved.

Reply:
1. Resolved.
3. I think is out of scope (although could mean `DumbConsoleLogger` would be a special case of `StreamLogger`)